src/components/withCurrentReportID.tsx

```diff
@@ -39,7 +39,21 @@ function CurrentReportIDContextProvider(props: CurrentReportIDContextProviderPro
      */
     const updateCurrentReportID = useCallback(
         (state: NavigationState) => {
-            setCurrentReportID(Navigation.getTopmostReportId(state) ?? '');
+            const reportID = Navigation.getTopmostReportId(state) ?? '';
+
+            /**
+             * This is to make sure we don't set the undefined as reportID when
+             * switching between chat list and settings tab. The settings tab
+             * includes multiple screens and we don't want to set the reportID
+             * to falsy value when switching between them.
+             *
+             * Doing so avoids an unnecessary re-render of `useReportIDs`.
+             */
+            const params = state?.routes?.[state.index]?.params;
+            if (params && 'screen' in params && typeof params.screen === 'string' && params.screen.indexOf('Settings_') !== -1) {
```

Comment:
```suggestion
            if (params && typeof params.screen === 'string' && params.screen.indexOf('Settings_') !== -1) {
```

Reply:
@hayata-suenaga We have to check for `screen` in params and without doing so, Typescript issues an error:

 "Property screen doesn't exist on type object"

I have followed this approach from other parts of the codebase like `getTopmostWorkspacesCentralPaneName.ts`.